bc-gh-2.3.2: new package

gavinhoward opened the pull request:
This is an implementation of POSIX bc and Unix dc with GNU extensions.
The webpage is at https://github.com/gavinhoward/bc .

I think that this bc could eventually replace the use of the GNU bc as
sys-devel/bc, since it can also build the Linux kernel, but for now, I
would be happy to just have it as an option in Gentoo.

Signed-off-by: Gavin D. Howard <yzena.tech@gmail.com>

juippis commented on sci-calculators/bc-gh/bc-gh-1.2.6.ebuild:
```diff
@@ -0,0 +1,32 @@
+
+LICENSE="BSD-2"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~hppa ~ia64 ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86"
```
You can't add KEYWORDS for systems you haven't tested this on, and even with some more exotic arches, you need to belong in the arch team before you can add support for them.

gavinhoward replied:
Eh, sorry about that. I just copied and pasted. My bad.

Note that I kept the `KEYWORDS` that I did because my bc builds and runs on those architectures. See http://pkg.musl.cc/bc-1.2.6/ for more information.

juippis commented on sci-calculators/bc-gh/bc-gh-1.2.6.ebuild:
```diff
@@ -0,0 +1,32 @@
+S="${WORKDIR}/bc-${PV}"
+
+DEPEND=""
+RDEPEND="${DEPEND}"
```
This seems rather useless. You can leave these variables empty, or remove them altogether if nothing goes into them.

gavinhoward replied:
Removed.

juippis commented on sci-calculators/bc-gh/bc-gh-1.2.6.ebuild:
```diff
@@ -0,0 +1,32 @@
+	einfo "Running ${configure_call[@]}"
+	EXECSUFFIX="-gh" \
+	PREFIX="${EPREFIX}/usr" \
+	"${configure_call[@]}" || die
```
Is all this complexity really necessary? Why wouldn't a single-liner 

`EXECSUFFIX="-gh" PREFIX="${EPREFIX}/usr" ./configure.sh -G || die` work here?

Of course you can export those variables into ENV before calling configure.sh if that's needed.

gavinhoward replied:
I had checked a recently-update ebuild for how it did its configure step with a custom `configure.sh` script, and that is what it had. I thought that I would need to follow that. Sorry about that.

Changed to the one-liner.

juippis commented on sci-calculators/bc-gh/metadata.xml:
```diff
@@ -0,0 +1,22 @@
+	<maintainer type="person">
+		<name>Gavin D. Howard</name>
+		<email>yzena.tech@gmail.com</email>
+	</maintainer>
```
You need to add a project or a developer to commit for you. It's usually the proxy-maint team, check from other proxy-maintained ebuilds for the correct syntax.

gavinhoward replied:
Added proxy-maint team.

juippis commented:
Also bear in mind lately new packages are being merged into ::gentoo slowly, so maybe GURU overlay would be a better starting point for this, for now.

gavinhoward commented:
@juippis thank you for your comments. I have fixed them, but I will take your advice and see if the GURU overlay wants this instead.

juippis commented:
I mean, this is a good PR, don't close it :P 

I'm just saying new packages seem to have the smallest priority amongst proxy-maint team, so if nothing happens for a while, you could try through GURU.

gavinhoward commented:
Ah, okay. :) Thank you for clarifying. Reopening, and will fix the comments addressed.

gavinhoward commented:
I fixed the comments. I also updated to 1.2.7, since it is out now. (All it has are fixes for the build on Arch Linux; there are no changes to source or the Gentoo build.)

gavinhoward commented:
Someone spotted a harmless (still worked, just copied extra data) bug in the `locale_install.sh` script, so I released 1.2.8 to fix it.

gavinhoward commented:
The `bc` has been updated to `2.0.0`, which brings several improvements. The best improvement is performance; this `bc` now outperforms the GNU `bc`.

juippis commented on sci-calculators/bc-gh/bc-gh-2.3.1.ebuild:
```diff
@@ -0,0 +1,19 @@
+
+LICENSE="BSD-2"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~ia64 ~ppc ~ppc64 ~s390 ~sh ~x86"
```
You can only add ~amd64 or ~x86 for new packages, and even ~x86 requires you to test it on x86 architechture. For the rest, you need to belong to these arch teams before you can add the keywords.

gavinhoward replied:
Fixed.

juippis commented:
Now your commit is missing the required sign-off. Please also describe new package in few words / sentences in the git message box (so below summary).

You can automate signing off with your git settings, or by using repoman to commit and adding DCO_SIGNED_OFF_BY into your make.conf.

gavinhoward commented:
I hope I managed to do what you asked.

juippis commented:
```
 * QA Notice: Files built without respecting LDFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/bin/bc-gh
 * QA Notice: Pre-stripped files found:
 * /usr/bin/bc-gh
```
As upstream, you should be able to deal with this issue. Can you patch it so it works in Gentoo?

You can replicate these errors with these settings, 
https://wiki.gentoo.org/wiki/Package_testing#make.conf

gavinhoward commented:
@juippis I have not been able to reproduce the errors. I have been investigating the errors to figure out exactly what they mean. I have an idea, but I haven't been able to confirm it. Anyway, I will try to figure out how to fix them.

gavinhoward commented:
I have figured out how to fix the problem, and I put out a point release with a fix to the build system.
